api/src/main/java/io/strimzi/api/kafka/model/listener/v2/ArrayOrObjectKafkaListeners.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.api.kafka.model.listener.v2;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonToken;
+import com.fasterxml.jackson.core.ObjectCodec;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonDeserializer;
+import com.fasterxml.jackson.databind.JsonSerializer;
+import com.fasterxml.jackson.databind.SerializerProvider;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import io.strimzi.api.kafka.model.Constants;
+import io.strimzi.api.kafka.model.listener.KafkaListeners;
+import io.strimzi.crdgenerator.annotations.Alternation;
+import io.strimzi.crdgenerator.annotations.Alternative;
+import io.sundr.builder.annotations.Buildable;
+
+import java.io.IOException;
+import java.io.Serializable;
+import java.util.List;
+
+@Buildable(
+        editableEnabled = false,
+        builderPackage = Constants.FABRIC8_KUBERNETES_API
+)
+@JsonDeserialize(using = ArrayOrObjectKafkaListeners.Deserializer.class)
+@JsonSerialize(using = ArrayOrObjectKafkaListeners.Serializer.class)
+@Alternation
+public class ArrayOrObjectKafkaListeners implements Serializable {
+    private static final long serialVersionUID = 1L;
+
+    private final List<GenericKafkaListener> listValue;
+    private final KafkaListeners objectValue;
+
+    public ArrayOrObjectKafkaListeners(List<GenericKafkaListener> listValue)   {
+        this.listValue = listValue;
+        this.objectValue = null;
+    }
+
+    public ArrayOrObjectKafkaListeners(KafkaListeners objectValue)   {
+        this.listValue = null;
+        this.objectValue = objectValue;
+    }
+
+    @Alternative()
+    public List<GenericKafkaListener> getListValue() {
+        return listValue;
+    }
+
+    @Alternative()
+    public KafkaListeners getObjectValue() {
+        return objectValue;
+    }
+
+    /**
+     * Convenience method which returns either the new listener format if set, or converted old format.
+     *
+     * @return  List of new listeners
+     */
+    public List<GenericKafkaListener> newOrConverted()  {
+        if (listValue != null)  {
+            return listValue;
+        } else {
```

Comment:
this else can be removed, since we return in the if

Reply:
I prefer to finish the else since it is more readable.